crates/bevy_reflect/src/enums/enum_trait.rs

```diff
@@ -92,16 +95,34 @@ use std::sync::Arc;
 pub trait Enum: PartialReflect {
     /// Returns a reference to the value of the field (in the current variant) with the given name.
     ///
-    /// For non-[`VariantType::Struct`] variants, this should return `None`.
-    fn field(&self, name: &str) -> Option<&dyn PartialReflect>;
+    /// # Errors
+    ///
+    /// - [`ReflectFieldError::NotFound`] if the field does not exist or is ignored.
+    /// - [`ReflectFieldError::NotFound`] if the field is not a [`VariantType::Struct`].
+    /// - [`ReflectFieldError::Readonly`] if the field is marked as readonly.
```

Comment:
Does this means we can't access read-only fields even as immutable?

Reply:
Oops, I don't think that line was supposed to be there. And I think I made that mistake on the other immutable field method as well 😅 

I'll fix this!